Regression: unresolved import

A [stable-to-beta crater report](https://gist.github.com/alexcrichton/ab516d4d73470c9da0884780d716fd2b) shows regressions related to 

```
error[E0432](B: unresolved import `glium::glutin::VirtualKeyCode`(B
  (B--> (Bsrc/graphics.rs:80:9(B
   (B|(B
80(B (B| (B    use (Bglium::glutin::VirtualKeyCode as Gvkc(B;(B
   (B| (B        (B^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^(B Could not find `glutin` in `glium`(B

```

* [jamkit-0.3.1](https://crates.io/crates/jamkit) ([before](https://tools.taskcluster.net/task-inspector/#nwqm5CzhTwCqa1KZ2FPUbQ)) ([after](https://tools.taskcluster.net/task-inspector/#i9NJKAYkT5usQvG04StJpg))

I'm opening an issue to ensure that we're tracking these regressions and they're either expected (so we can close), this is a dupe (so we can close), or we can fix (and then close!)

cc @rust-lang/compiler, @jseyfried

--- a/src/librustc_resolve/resolve_imports.rs
+++ b/src/librustc_resolve/resolve_imports.rs
@@ -197,7 +197,8 @@ impl<'a> Resolver<'a> {
         // If the resolution doesn't depend on glob definability, check privacy and return.
         if let Some(result) = self.try_result(&resolution, ns) {
             return result.and_then(|binding| {
-                if self.is_accessible(binding.vis) && !is_disallowed_private_import(binding) {
+                if self.is_accessible(binding.vis) && !is_disallowed_private_import(binding) ||
+                   binding.is_extern_crate() { // c.f. issue #37020
                     Success(binding)
                 } else {
                     Failed(None)

--- a/src/test/run-pass/issue-37020.rs
+++ b/src/test/run-pass/issue-37020.rs
@@ -1,0 +1,25 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![allow(private_in_public)]
+
+mod foo {
+    pub mod bar {
+        extern crate core;
+    }
+}
+
+mod baz {
+    pub use foo::bar::core;
+}
+
+fn main() {
+    baz::core::cell::Cell::new(0u32);
+}

